src/librustdoc/html/render/mod.rs

```diff
@@ -2410,3 +2429,206 @@ fn collect_paths_for_type(first_ty: clean::Type, cache: &Cache) -> Vec<String> {
+    let call_locations = match cx.shared.call_locations.get(&key) {
+        Some(call_locations) if call_locations.len() > 0 => call_locations,
+        _ => {
+            return;
+        }
+    };
+
+    // Generate a unique ID so users can link to this section for a given method
+    let id = cx.id_map.borrow_mut().derive("scraped-examples");
+    write!(
+        w,
+        r##"<div class="docblock scraped-example-list">
+          <h1 id="scraped-examples" class="small-section-header">
+             <a href="#{}">Examples found in repository</a>
+          </h1>"##,
+        id
+    );
+
+    // Generate the HTML for a single example, being the title and code block
+    let write_example = |w: &mut Buffer, (path, call_data): (&PathBuf, &CallData)| -> bool {
+        let contents = match fs::read_to_string(&path) {
+            Ok(contents) => contents,
+            Err(err) => {
+                let span = item.span(tcx).inner();
+                tcx.sess
+                    .span_err(span, &format!("failed to read file {}: {}", path.display(), err));
+                return false;
+            }
+        };
+
+        // To reduce file sizes, we only want to embed the source code needed to understand the example, not
+        // the entire file. So we find the smallest byte range that covers all items enclosing examples.
+        assert!(call_data.locations.len() > 0);
+        let min_loc =
+            call_data.locations.iter().min_by_key(|loc| loc.enclosing_item.byte_span.0).unwrap();
+        let (byte_offset, _) = min_loc.enclosing_item.byte_span;
+        let (line_offset, _) = min_loc.enclosing_item.line_span;
+        let byte_ceiling =
+            call_data.locations.iter().map(|loc| loc.enclosing_item.byte_span.1).max().unwrap();
+
+        // The output code is limited to that byte range.
+        let contents_subset = &contents[(byte_offset as usize)..(byte_ceiling as usize)];
+
+        // The call locations need to be updated to reflect that the size of the program has changed.
+        // Specifically, the ranges are all subtracted by `byte_offset` since that's the new zero point.
+        let (byte_ranges, line_ranges): (Vec<_>, Vec<_>) = call_data
+            .locations
+            .iter()
+            .map(|loc| {
+                let (byte_lo, byte_hi) = loc.call_expr.byte_span;
+                let (line_lo, line_hi) = loc.call_expr.line_span;
+                (
+                    (byte_lo - byte_offset, byte_hi - byte_offset),
+                    (line_lo - line_offset, line_hi - line_offset),
+                )
+            })
+            .unzip();
+
+        let (init_min, init_max) = line_ranges[0];
+        let line_range = if init_min == init_max {
+            format!("line {}", init_min + line_offset + 1)
+        } else {
+            format!("lines {}-{}", init_min + line_offset + 1, init_max + line_offset + 1)
+        };
+
+        let edition = cx.shared.edition();
```

Comment:
This is another thing that breaks because you don't have access to the original span - the edition can be different between the example and the library it uses. (There's actually lots of subtleties around this related to proc-macros, but getting the right one per-crate would make me happy for now.) Maybe serialize the edition along with the range?

Reply:
Good idea, I've added serialization of the edition.